packages/angular-cli/models/webpack-build-utils.ts

```diff
@@ -109,3 +109,18 @@ export function extraEntryParser(
       return extraEntry;
     });
 }
+
+const hashFormats: { [option: string]: any } = {
+  'none':     { chunk: '',             extract: '',                  file: '[name]'    },
+  'media':    { chunk: '',             extract: '',                  file: '[hash:20]' },
```

Comment:
Regarding hash size, I think it's better to be consistent with the rest of the hashes already present so 20 sounds good to me. 

The question of including the name is less obvious for me though. Having the name would be a change in the previous behaviour, but if we look at it with fresh eyes I think it's simply a better option since now everything that's processed follows the same pattern of `[name].[hash].ext`.

Some users are confused by the multitude of files suddenly propping up in their dist folder and that would help them make sense of it. It also adds a bit more collision protection.

So my vote goes for `[name].[hash:20]`. I think `[name]` should be in the common config instead of here though (like the others).